nixos/modules/config/xdg/terminal-exec.nix

```
@@ -0,0 +1,58 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.xdg.terminal-exec;
+  inherit (lib) mkIf mkEnableOption mkOption mkPackageOption types;
+in
+{
+  meta.maintainers = with lib.maintainers; [ Cryolitia ];
+
+  ###### interface
+
+  options = {
+
+    xdg.terminal-exec = {
+      enable = mkEnableOption "xdg-termonal-exec, the [proposal](https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/46) for XDG terminal execution utility";
+      package = mkPackageOption pkgs "xdg-terminal-exec" { };
+      settings = mkOption {
+        type = with types; attrsOf (listOf str);
+        default = { };
+        description = ''
+          Configuration options for XDG terminal execution utility.
+
+          The keys are the desktop environments, and the values are a list of terminals' [desktop file IDs](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id) to try.
+
+          To define your current desktop environment, run `echo $XDG_CURRENT_DESKTOP`.
+
+          The default value is used when the desktop environment is not found in the configuration.
```

```suggestion
          Configuration options for the Default Terminal Execution Specification.

          The keys are the desktop environments that are matched (case-insensitively) against `$XDG_CURRENT_DESKTOP`,
          or `default` which is used when the current desktop environment is not found in the configuration.
          The values are a list of terminals' [desktop file IDs](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id) to try in order of decreasing priority.
```

I intentionally decreased the amount of whitespace, we don't need everything to be on a new line.